### DISCUSSION TITLE
Static export with flexsearch

### DISCUSSION THREAD

**Discussion by Skreeauk** [fuma-nama/fumadocs#764] [Q&A] [answered]
I'm looking to export static to reduce bandwith etc, however the flex search uses searchParams so it raises an error there. Is there another solution beside using Algolia?

**Comment by fuma-nama** [comment, marked as answer]
I was thinking to export a static search map and index it on client side, but I didn't support it because:

- We want full-text search, but it also means that users need to download **the content of all pages** if we want to index them on client side
- I still think that handling search on server is the best option

If the free tier of Algolia isn't enough, you can:

- Host the search API on another server (hosted on anywhere you want, e.g. Railway): https://fumadocs.vercel.app/docs/headless/search/flexsearch#headless
- Choose other search solutions, like https://orama.com or Elasticsearch. See https://fumadocs.vercel.app/docs/ui/search#other-solutions to replace the default search dialog

The information of search indexes is called Structured Data, on Fumadocs MDX it's generated with: https://fumadocs.vercel.app/docs/mdx/search-index.
You can use it to integrate with other search solutions

**Reply by Skreeauk** [reply]
I see. Algolia free tier will be enough for my use case, I just wanted to know if there were other alternatives beside Algolia.

Thanks for the quick response.

**Comment by Skreeauk** [comment]
@fuma-nama 

So I got algolia working, but how do I add keywords to it? 

I tried looking and I think I'm supposed to pass `extra_data`, run `setIndexSettings` then `updateDocuments` like what `sync` does. But I find that I could not find the additional frontmatter that I added in `source` in the json that was generated from `buildSearchIndex`

**Reply by fuma-nama** [reply]
Oh you are right, will support it later.

Content Collections can solve it if it's an urgent matter.

**Reply by Skreeauk** [reply]
Thanks for making it a feature later. Cheers

**Reply by fuma-nama** [reply]
It should be supported now

**Reply by Skreeauk** [reply]
Thanks, will check it later.

Also a sidenote, I suggest a note to install algoliasearch ver 4 to use your code, as ver 5 came out ~2 weeks ago and `initIndex` is removed, and some other stuff https://www.algolia.com/doc/libraries/javascript/v5/upgrade/

**Comment by Shubham-EV** [comment]
I am using flex search and trying to use tag filters, but it doesn't seem to work. Also, [here](https://fumadocs.vercel.app/docs/headless/search/flexsearch#tag-filter) I get an error: "page.data.tag doesn't exist." Can you tell me how to configure it using flex search? Currently, on your repo, it is configured using Algolia search and
[here](https://fumadocs.vercel.app/docs/ui/search#tag-filter) can you tell me what value refers to ?
